src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -162,164 +195,69 @@ void SysNewton::update_translate(ACtxPhysics& rCtxPhys, ACtxNwtWorld& rCtxWorld)
-            break;
-        case EShape::Capsule:
-            // TODO
-            break;
-        case EShape::Cylinder:
-            pNwtCollider = NewtonCreateCylinder(pNwtWorld, 1, 1, 2, 0, nullptr);
-            break;
-        default:
-
-            break;
-        }
-
-        if (nullptr != pNwtCollider)
-        {
-            if (rCtxWorld.m_nwtColliders.contains(ent))
-            {
-                // Replace existing Newton collider component
-                rCtxWorld.m_nwtColliders.get(ent).reset(pNwtCollider);
-            }
-            else
-            {
-                // Add new Newton collider component
-                rCtxWorld.m_nwtColliders.emplace(ent, pNwtCollider);
-            }
-
-        }
-    }
-}
-
 using Corrade::Containers::ArrayView;
 
 void SysNewton::update_world(
         ACtxPhysics& rCtxPhys,
         ACtxNwtWorld& rCtxWorld,
         float timestep,
-        ArrayView<ACtxPhysInputs> inputs,
-        acomp_storage_t<ACompHierarchy> const& rHier,
-        acomp_storage_t<ACompTransform>& rTf,
-        acomp_storage_t<ACompTransformControlled>& rTfControlled,
-        acomp_storage_t<ACompTransformMutable>& rTfMutable) noexcept
+        ACtxSceneGraph const& rScnGraph,
+        acomp_storage_t<ACompTransform>& rTf) noexcept
 {
-
-    NewtonWorld const* pNwtWorld = rCtxWorld.m_nwtWorld.get();
-
-    // Iterate rigid bodies that don't have a NewtonBody
-
-    entt::basic_view<ActiveEnt, entt::get_t<ACompPhysBody>, entt::exclude_t<ACompNwtBody_t>> view{rCtxPhys.m_physBody, rCtxWorld.m_nwtBodies};
-
-    for (ActiveEnt ent : view)
-    {
-        create_body(rCtxPhys, rCtxWorld, rHier, rTf, rTfControlled,
-                    rTfMutable, ent, pNwtWorld);
-    }
+    NewtonWorld const* pNwtWorld = rCtxWorld.m_world.get();
 
     // Apply changed velocities
-    for (ACtxPhysInputs &rInputs : inputs)
+    for (auto const& [ent, vel] : std::exchange(rCtxPhys.m_setVelocity, {}))
     {
-        for (auto const& [ent, vel] : std::exchange(rInputs.m_setVelocity, {}))
-        {
-            NewtonBodySetVelocity(rCtxWorld.m_nwtBodies.get(ent).get(), vel.data());
-        }
-    }
+        BodyId const bodyId     = rCtxWorld.m_entToBody.at(ent);
+        NewtonBody const *pBody = rCtxWorld.m_bodyPtrs[bodyId].get();
 
-    // Expose force and torque inputs to Newton callbacks
-    // This swaps their internal pointers with instances accesible from the
-    // cb_force_torque callback function.
-    // The alternative is pointing to the storages from the callback, which
-    // requires additional indirection
-    rCtxWorld.m_forceTorqueIn.resize(inputs.size());
-    for (int i = 0; i < inputs.size(); i ++)
-    {
-        std::swap(rCtxWorld.m_forceTorqueIn[i].m_force, inputs[i].m_physNetForce);
-        std::swap(rCtxWorld.m_forceTorqueIn[i].m_torque, inputs[i].m_physNetTorque);
+        NewtonBodySetVelocity(pBody, vel.data());
     }
 
+    rCtxWorld.m_pTransform = &rTf;
```

Comment:
i recommend std::addressof() to get the address of things.

The & is fine, but it's hard to notice if you're reading quickly.